src/main/java/com/amazon/opendistroforelasticsearch/ad/model/AnomalyDetectorExecutionInput.java

```diff
@@ -98,6 +107,9 @@ public static AnomalyDetectorExecutionInput parse(XContentParser parser, String
                     break;
             }
         }
+        if (Strings.isNotEmpty(adId)) {
+            detectorId = adId;
+        }
```

Comment:
this looks a little bit wired, if the adId == "", then we will apply null to detectorId. Could you explain the difference of "" and null here?

Reply:
No difference, this is to prevent some user wrong input. The detector id should not be null or "".  If the user input "" , we still consider that no detector id provided.